heareval/tasks/speech_commands.py

```diff
@@ -2,38 +2,183 @@
 """
 Pre-processing pipeline for Google Speech Commands
 """
-# from heareval.tasks.coughvid import DownloadCorpus, ExtractCorpus
 import os
-import subprocess
+import re
+from functools import partial
+from glob import glob
 
+import heareval.tasks.config.speech_commands as config
 import luigi
 import numpy as np
-
-import heareval.tasks.config.speech_commands as config
-import heareval.tasks.util.audio as audio_util
-import heareval.tasks.util.luigi as luigi_util
-from heareval.tasks.util.luigi import DownloadCorpus, WorkTask, ExtractArchive
-
+import pandas as pd
+from heareval.tasks.util.luigi import (
+    PROCESSMETADATACOLS,
+    DownloadCorpus,
+    ExtractArchive,
+    FinalizeCorpus,
+    MetadataVocabulary,
+    MonoWavTrimCorpus,
+    ResampleSubCorpus,
+    SplitTrainTestCorpus,
+    SplitTrainTestMetadata,
+    SubsampleCorpus,
+    WorkTask,
+    ensure_dir,
+    filename_to_int_hash,
+    slugify_file_name,
+    which_set,
+)
 
 # Set the task name for all WorkTasks
 WorkTask.task_name = config.TASKNAME
 
 
+class ExtractArchive(ExtractArchive):
+    def requires(self):
+        return {
+            "download": DownloadCorpus(url=config.DOWNLOAD_URL, outfile="corpus.tar.gz")
+        }
+
+
+class ConfigureProcessMetaData(WorkTask):
+    """
+    This config is data dependent and has to be set for each data
+    """
+
+    def requires(self):
+        return ExtractArchive(infile="corpus.tar.gz")
+
+    def run(self):
+        # Get relative path of the audio files
+        # This file can also be built with the metadata file for the dataset
+        # in case the metadata is provided
+        process_metadata = pd.DataFrame(
+            glob(os.path.join(self.requires().workdir, "[!_]*/*.wav")),
```

Comment:
As part of the `ConfigureProcessMetaData`, the metadata file is made which pulls up all the files which will be used for training and testing. Since this is data specific as different data can have different folder structures and different ways to get these files, we get the relative path of those files and add it to the metadata so that going ahead these files can be treated as just training points. @jorshi To make background data to not be part of the training data, I have added the regex `[!_]*/*.wav` so that the `_background` folder is not used while building the training points list. I have made an assumption that the files in this will not be part of the training or testing data. But as I understand if we need to make another category as `background` by doing some processing, we can do it here as well. This is kind of the design of the flow that makes all these decisions in this class only leaving the subsequent tasks to do the heavy lifting of executing the process with the data-specific decisions made by this class.